rpc/grpc/server/services/blockresultservice/service.go

```diff
@@ -0,0 +1,61 @@
+package blockresultservice
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cometbft/cometbft/libs/log"
+	sm "github.com/cometbft/cometbft/state"
+	"github.com/cometbft/cometbft/store"
+
+	brs "github.com/cometbft/cometbft/proto/tendermint/services/block_results/v1"
+)
+
+type blockResultsService struct {
+	stateStore sm.Store
+	blockStore *store.BlockStore
+	logger     log.Logger
+}
+
+// New creates a new CometBFT block results service server.
+func New(bs *store.BlockStore, ss sm.Store, logger log.Logger) brs.BlockResultsServiceServer {
+	return &blockResultsService{stateStore: ss, blockStore: bs, logger: logger}
+}
+
+// GetBlockResults returns the block results of the requested height.
+// If no height is given, the block results for the latest height are returned.
+func (s *blockResultsService) GetBlockResults(_ context.Context, req *brs.GetBlockResultsRequest) (*brs.GetBlockResultsResponse, error) {
+	logger := s.logger.With("endpoint", "GetBlockResults")
+	height := req.Height
+	latestHeight := s.blockStore.Height()
+	if req.Height > latestHeight || req.Height < 0 {
+		logger.Error("Failed to validate height from BlockResults request", "height", height)
+		return nil, status.Error(codes.InvalidArgument, "Invalid request: height must be between 0 and the last effective height.")
+	} else if req.Height == 0 {
+		height = latestHeight
+	}
+
+	res, err := s.stateStore.LoadFinalizeBlockResponse(height)
+	if err != nil {
+		logger.Error("Error fetching BlockResults", "height", height, "err", err)
+		return nil, status.Errorf(codes.Internal, "Internal server error: cannot fetch BlockResults for height %d", height)
```

Comment:
```suggestion
		return nil, status.Error(codes.Internal, "Internal server error")
```

Since the error and the height are being logged, operators can dig into what's going on here. As a stretch goal, I'd recommend looking into using this request tracing approach I've implemented here: https://github.com/cometbft/cometbft/pull/1142/files#diff-2cdfb60e6f68e99a22b64a5d79cb26e00adb76e670b0f6d144b7912bf1ae2354R84

That way, if a user asks the operator about why they're getting internal server errors, the operator will have logs to link their specific requests to.